### ISSUE TITLE
HermeneutiX: allow multi-language configuration of syntactic functions

### ISSUE THREAD

**Issue by CarstenWickner** [scientific-tool-set/scitos#14]
As a HermeneutiX user I want to be able to define the same syntactic functions in multiple languages, in order to be able to export the same analysis for varying target languages/contexts.

One option would be to allow switching of the "Origin Language" even after the analysis was started.
Another option could be to define multiple versions of the assigned "Code" (and "Name"?) values and switch between those sets/profiles.

Thanks to Phil (@j2l) for the suggestion.

**Comment by CarstenWickner** [comment]
I don't think switching between origin languages is feasible here, since we would need to get a one-to-one mapping between the included functions (worst case: when switching between them).
I'd rather go for a single more complex origin language that contains the same syntactic function in multiple languages/set/profiles.
For that however, the configuration would need to be completely re-designed.
In short: I'm open for design suggestions. :)

**Comment by CarstenWickner** [comment]
I'm looking into this now. From a first glance, I'm planning for the following changes:

1. Extension of the origin language data model to incorporate "profiles"
    - Each syntactic function should have a new "id" field.
    - Each profile contains mappings from "id" to the respective "Code"/"Name"/"Description"
    - One profile is to be marked as the default.
    - If a non-default profile does not contain a mapping for a given id, the values from the default profile are being used instead.
2. Adjust XML file structure to incorporate profiles.
    - Similar to data model structure: separate list of profiles should be included.
    - New Structure should ideally be backwards compatible, i.e. new structure should still be possible to be opened with version 2.0.0 – just discarding non-default profiles on "Save".
    - When saving in new structure, currently displayed profile should be saved as the active (not default) one.
3. Convert system-provided "Greek" and "Hebrew" models into new structure with "English" and "German" profiles.
4. Offer profile selection on Text Input view below origin language selection.
5. Add View menu item "Select Display Profile: Syntactic Functions" with sub items for each available profile.
6. Add configuration of profiles for syntactic functions of a single origin language to user settings/preferences.
    - Each profile has a "Name".
    - One profile is marked as the "Default Profile" for the associated origin language.
    - The default profile must contain mappings for all syntactic functions.
    - If a non-default profile is missing a mapping, the user should have the option to apply the default profile's mappings for the missing ones in order to be allowed to save the changes anyway.
    - Ideally, also implement #24 (keyboard navigation while editing TreeTable contents) here in order to make this actually usable.
7. Add Edit menu item "Configure Display Profiles: Syntactic Functions"
    - SImilar to user settings, but only for altering the currently displayed project's origin language model's profiles.

I expect this will take me a while.

**Comment by j2l** [comment]
I believe you. Courage
